### PR TITLE
Windows install fix

### PR DESCRIPTION
**Pull request by varjolintu**
Kills keepassxc-proxy.exe during install/uninstall.

## Motivation and context
Solves https://github.com/keepassxreboot/keepassxc/issues/1521.

## How has this been tested?
Manually.

## Types of changes
- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

**Comment by droidmonkey** [comment]
This would fix it for NSIS, is there a complimentary command for WIX?

**Comment by varjolintu** [comment]
I'll assume the correct command for WIX is the following:
```
<InstallExecuteSequence>
    <Custom Action="KillProxy" After="InstallInitialize" />
    <Custom Action='KillProxy' Before='InstallFinalize'>(NOT UPGRADINGPRODUCTCODE) AND (REMOVE="ALL")</Custom>
</InstallExecuteSequence>

<Property Id="WixQuietExecCmdLine" Value="Taskkill /IM keepassxc-proxy.exe /F"/>
<CustomAction Id="KillProxy" BinaryKey="WixCA" DllEntry="WixQuietExec" Execute="immediate" Return="ignore"/>
```

I'll try if it works soon.

**Comment by varjolintu** [comment]
Fix for WIX committed. Rebased.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -362,6 +362,8 @@ if(MINGW)
   set(CPACK_NSIS_MUI_UNWELCOMEFINISHPAGE_BITMAP "${CPACK_NSIS_MUI_WELCOMEFINISHPAGE_BITMAP}")
   set(CPACK_NSIS_CREATE_ICONS_EXTRA "CreateShortCut '$SMPROGRAMS\\\\$STARTMENU_FOLDER\\\\${PROGNAME}.lnk' '$INSTDIR\\\\${PROGNAME}.exe'")
   set(CPACK_NSIS_DELETE_ICONS_EXTRA "Delete '$SMPROGRAMS\\\\$START_MENU\\\\${PROGNAME}.lnk'")
+  set(CPACK_NSIS_EXTRA_INSTALL_COMMANDS "ExecWait 'Taskkill /IM keepassxc-proxy.exe /F'")
```
Comment (droidmonkey) [inline src/CMakeLists.txt]: This needs to be ```CPACK_NSIS_EXTRA_PREINSTALL_COMMANDS``` so they execute before any files are written and need to add an extra command to kill keepassxc.exe itself. Here is the desired entry:

```
set(CPACK_NSIS_EXTRA_PREINSTALL_COMMANDS
          "ExecWait 'Taskkill /IM keepassxc-proxy.exe /F'
          ExecWait 'Taskkill /IM keepassxc.exe /F'")
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -362,6 +362,8 @@ if(MINGW)
...
+  set(CPACK_NSIS_EXTRA_UNINSTALL_COMMANDS "ExecWait 'Taskkill /IM keepassxc-proxy.exe /F'")
```
Comment (droidmonkey) [inline src/CMakeLists.txt]: Need to add keepassxc.exe, same as above

File: share/windows/wix-template.xml [hunk 3]
```diff
@@ -47,5 +47,14 @@
         <DirectoryRef Id="TARGETDIR">
             <Directory Id="ProgramMenuFolder" />
         </DirectoryRef>
+
+        <Property Id="WixQuietExecCmdLine" Value='"Taskkill" /IM keepassxc-proxy.exe /F'/>
```
Comment (droidmonkey) [inline share/windows/wix-template.xml]: Need to add ```/IM keepassxc.exe``` just like NSIS

File: src/CMakeLists.txt (hunk 2 above)
Comment (phoerious) [reply src/CMakeLists.txt]: Windows is case-insensitive, but I would still use the correct KeePassXC.exe, not keepassxc.exe.

**Comment by varjolintu** [comment]
I'll do the rebase after I've tested that the new additions work correctly.

**Comment by varjolintu** [comment]
NSIS works. WIX doesn't yet.

**Comment by varjolintu** [comment]
Now everything works OK. Rebased.

**Pull request merged**